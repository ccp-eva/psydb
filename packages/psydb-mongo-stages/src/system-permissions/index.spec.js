'use strict';
var { SystemPermissionStages } = require('./');

describe('system-permission-stages', () => {
    
    test('root access has no additional stages', () => {
        var stages = SystemPermissionStages({
            collection: 'subject',
            permissions: { hasRootAccess: true }
        })
        expect(stages).toEqual([]);
    });

    test('non-root with one research group has proper stages', () => {
        var stages = SystemPermissionStages({
            collection: 'subject',
            permissions: {
                hasRootAccess: false,
                //projectedResearchGroupIds: [ 'gid_1' ],
                //byResearchGroupId: {
                //    'gid_1': { canReadSubjects: true }
                //},
                researchGroupIdsByCollection: {
                    'subject': { read: [ 'gid_1' ], write: [] }
                }
            }
        });

        //console.dir(stages, { depth: null });
        expect(stages).toMatchSnapshot()
    })

    test('root access with forced group has proper stages', () => {

    })

})
