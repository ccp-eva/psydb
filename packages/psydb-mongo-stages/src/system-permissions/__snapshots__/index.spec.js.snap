// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`system-permission-stages non-root with one research group has proper stages 1`] = `
Array [
  Object {
    "$match": Object {
      "$expr": Object {
        "$gt": Array [
          Object {
            "$size": Object {
              "$ifNull": Array [
                Object {
                  "$setIntersection": Array [
                    "$scientific.state.systemPermissions.accessRightsByResearchGroup.researchGroupId",
                    Array [
                      "gid_1",
                    ],
                  ],
                },
                Array [],
              ],
            },
          },
          0,
        ],
      },
    },
  },
]
`;
